### PR TITLE
Run RPCs related to Solana tx fee estimation concurrently

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/38925

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/browser/solana_tx_manager_unittest.cc [hunk 1]
```diff
@@ -766,7 +766,7 @@ TEST_F(SolanaTxManagerUnitTest, AddAndApproveTransaction) {
   ASSERT_TRUE(tx_meta3);
   EXPECT_TRUE(tx_meta3->tx()->fee_estimation());
   EXPECT_EQ(tx_meta3->tx()->fee_estimation()->base_fee, 5000U);
-  EXPECT_EQ(tx_meta3->tx()->fee_estimation()->compute_units, 69017U);
+  EXPECT_EQ(tx_meta3->tx()->fee_estimation()->compute_units, 69317U);
```
Comment (yrliou) [inline components/brave_wallet/browser/solana_tx_manager_unittest.cc:769]: I think it was a bug, when we use default fee per compute unit, we should still add 300 to it for priority fee instructions.

File: components/brave_wallet/browser/solana_tx_manager.cc [hunk 2]
```diff
@@ -927,119 +913,129 @@ void SolanaTxManager::OnGetAccountInfo(
...
-  // and unsigned transaction used in simulateTransaction.
   meta->tx()->message()->set_recent_blockhash(latest_blockhash);
   meta->tx()->message()->set_last_valid_block_height(last_valid_block_height);
-  const std::string base64_encoded_message =
-      meta->tx()->GetBase64EncodedMessage();
-  const std::string unsigned_tx = meta->tx()->GetUnsignedTransaction();
+  std::string base64_encoded_message = meta->tx()->GetBase64EncodedMessage();
+  std::string unsigned_tx = meta->tx()->GetUnsignedTransaction();
+  std::string chain_id = meta->chain_id();
+  bool is_partial_signed = meta->tx()->IsPartialSigned();
```
Comment (nvonpentz) [inline components/brave_wallet/browser/solana_tx_manager.cc:983]: const?

File: components/brave_wallet/browser/solana_tx_manager.cc [hunk 3]
```diff
@@ -927,119 +913,129 @@ void SolanaTxManager::OnGetAccountInfo(
                           mojom::SolanaProviderError::kSuccess, "");
 }
 
-void SolanaTxManager::GetEstimatedTxBaseFee(
```
Comment (nvonpentz) [inline components/brave_wallet/browser/solana_tx_manager.cc:930]: I think `GetEstimatedTxBaseFee` and `GetEstimatedTxBaseFeeCallback` are still leftover in .h

File: components/brave_wallet/browser/solana_tx_manager_unittest.cc (hunk 1 above)
Comment (nvonpentz) [reply components/brave_wallet/browser/solana_tx_manager_unittest.cc:769]: Agree

**Review by nvonpentz — APPROVED** [review]

**Pull request merged**